File: src/data/checkOrder.ts
import { tollGates, passagePoints, routeSerbest_S1A_to_S5A } from "./mapData";
import { backboneS5A_G1, backboneG1_Bridge_G5, backboneG5_Asia } from "./highwayBackbone";

const fullBackbone = [
  ...routeSerbest_S1A_to_S5A,
  ...backboneS5A_G1.slice(1),
  ...backboneG1_Bridge_G5.slice(1),
  ...backboneG5_Asia.slice(1),
];

function getSnapIdx(lat: number, lng: number) {
  let minIdx = 0;
  let minD = Infinity;
  for (let i = 0; i < fullBackbone.length; i++) {
    const d = (fullBackbone[i][0] - lat)**2 + (fullBackbone[i][1] - lng)**2;
    if (d < minD) { minD = d; minIdx = i; }
  }
  return minIdx;
}

const all = [...tollGates, ...passagePoints];
const sorted = all.map(p => ({ id: p.id, idx: getSnapIdx(p.lat, p.lng) }))
                 .sort((a, b) => a.idx - b.idx);

console.log(JSON.stringify(sorted, null, 2));

File: check_order.py
import json
import re

# Read markers from mapData.ts
with open('src/data/mapData.ts', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract IDs and Coords
markers = []
# Match tollGates
tg_matches = re.findall(r"id: \"([^\"]+)\",\s+name: \"[^\"]+\",\s+lat: ([\d.]+),\s+lng: ([\d.]+)", content)
for m in tg_matches:
    markers.append({'id': m[0], 'lat': float(m[1]), 'lng': float(m[2])})
# Match passagePoints
pp_matches = re.findall(r"id: \"([^\"]+)\",\s+name: \"[^\"]+\",\s+subtitle: \"[^\"]+\",\s+lat: ([\d.]+),\s+lng: ([\d.]+)", content)
for m in pp_matches:
    markers.append({'id': m[0], 'lat': float(m[1]), 'lng': float(m[2])})

# Read backbone (hard to parse TS, so I'll just mock the full list)
# Actually, I'll use the file directly
with open('src/data/highwayBackbone.ts', 'r', encoding='utf-8') as f:
    bb_content = f.read()
    # Extract ALL [lat, lng]
    bb_coords = re.findall(r"\[([\d.]+),\s*([\d.]+)\]", bb_content)
    backbone = [[float(c[0]), float(c[1])] for c in bb_coords]

def get_snap_idx(lat, lng):
    min_idx = 0
    min_d = float('inf')
    for i, (blat, blng) in enumerate(backbone):
        d = (blat - lat)**2 + (blng - lng)**2
        if d < min_d:
            min_d = d
            min_idx = i
    return min_idx

sorted_markers = []
for m in markers:
    sorted_markers.append({'id': m['id'], 'idx': get_snap_idx(m['lat'], m['lng'])})

sorted_markers.sort(key=lambda x: x['idx'])

print(json.dumps(sorted_markers, indent=2))
